lib/order.rb

```diff
@@ -0,0 +1,65 @@
+require_relative 'menu'
+
+class Order
+
+  attr_reader :basket
+
+  def initialize
+    @basket = Hash.new(0)
+  end
+
+  def add(dish, quantity)
+    dish_quantity_check(quantity)
```

Comment:
These helper methods (`dish_quantity_check` and `dish_in_menu?`) clean up the code base and communicate your intention nicely.

I also like the fact that you can increment the quantity of an item already in the order.

How about this as an alternative pattern?  It makes the `#add` method feel less procedural.

```ruby
def add(dish, quantity)
  @basket[dish] += quantity if valid_addition?(dish, quantity)
end

def valid_addition?(dish, quantity)
  valid_quantity?(quantity) && valid_dish?(dish)
end

def valid_quantity?(quantity)
  quantity > 0 ? true : fail 'Quantity must be greater than 0'
end

def valid_dish(dish)
  # assuming a menu of key-value pairs
  menu.keys.include?(dish) ? true : fail 'That dish is not on the menu'
end
```